node/src/components/linear_chain/state.rs

```diff
@@ -410,6 +413,91 @@ mod tests {
         );
     }
 
+    // Creates a new finality signature for a given block, adds it as pending and returns.
+    fn add_pending(
+        lc: &mut LinearChain,
+        block_hash: BlockHash,
+        era_id: EraId,
+        local: bool,
+    ) -> FinalitySignature {
+        let sig = FinalitySignature::random_for_block(block_hash, era_id.value());
+        let outcomes = lc.handle_finality_signature(Box::new(sig.clone()), !local);
+        assert!(matches!(&*outcomes, [Outcome::LoadSignatures(_)]));
+        sig
+    }
+
+    // Mark the creator of the signature as bonded.
+    fn mark_bonded(lc: &mut LinearChain, fs: FinalitySignature) {
+        let outcomes = lc.handle_is_bonded(None, Box::new(fs), true);
+        assert!(outcomes.is_empty());
+    }
+
+    #[test]
+    fn new_block_unvalidated_pending_sigs() {
+        let mut rng = TestRng::new();
+        let protocol_version = ProtocolVersion::V1_0_0;
+        let mut lc = LinearChain::new(protocol_version, 1u64, 1u64);
+        let block = Block::random(&mut rng);
+        let block_hash = *block.hash();
+        let block_era = block.header().era_id();
+        // Store some pending finality signatures
+        let _sig_a = add_pending(&mut lc, block_hash, block_era, true);
+        let _sig_b = add_pending(&mut lc, block_hash, block_era, false);
+        let _sig_c = add_pending(&mut lc, block_hash, block_era, false);
+
+        let execution_results = HashMap::new();
+        let outcomes = lc.handle_new_block(Box::new(block), execution_results);
+        // None of the signatures' creators have been confirmed to be bonded yet.
+        // We should not gossip/store/announce any signatures yet.
+        assert!(matches!(
+            &*outcomes,
+            [Outcome::StoreBlock(_), Outcome::StoreExecutionResults(_, _)]
```

Comment:
~Should we mark our own signature automatically as bonded, and also store it?~
~Or would that just unnecessarily complicate the code?~
It probably would — if in doubt I'm definitely for fewer code paths.

Reply:
Hmm, you're correct that we could do that but I am afraid that it would complicate the code too much (just like you said).